File: jj/faultdiag-source-20260906/annotation-app/frontend/src/types/faultTypeSuggestion.ts
export type FaultTypeSuggestionStatus = 'pending' | 'accepted' | 'rejected'

export interface FaultTypeSuggestion {
  id: number
  slice_window_id: number
  suggested_name: string
  suggested_description: string
  reason: string | null
  model: string | null
  status: FaultTypeSuggestionStatus
  accepted_fault_type_id: number | null
  created_at: string
  updated_at: string
}

export interface FaultTypeSuggestionListResponse {
  items: FaultTypeSuggestion[]
  total: number
  page: number
  page_size: number
}

export interface FaultTypeSuggestionAcceptPayload {
  name?: string | null
  description?: string | null
}

export interface FaultTypeSuggestionAcceptResponse {
  suggestion: FaultTypeSuggestion
  fault_type: {
    id: number
    name: string
    description: string
    created_at: string
    updated_at: string
  }
}

File: jj/code_9.24/annotation-app/backend/app/schemas/fault_type_suggestion.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field, field_validator

from app.schemas.fault_type import FaultTypeResponse

SuggestionStatus = Literal["pending", "accepted", "rejected"]


class FaultTypeSuggestionResponse(BaseModel):
    id: int
    slice_window_id: int
    suggested_name: str
    suggested_description: str
    reason: str | None
    model: str | None
    status: SuggestionStatus
    accepted_fault_type_id: int | None
    created_at: datetime
    updated_at: datetime


class FaultTypeSuggestionListResponse(BaseModel):
    items: list[FaultTypeSuggestionResponse]
    total: int
    page: int
    page_size: int


class FaultTypeSuggestionAcceptRequest(BaseModel):
    """Override-able fields when accepting a suggestion. If both are omitted the
    LLM-proposed name/description are used verbatim."""

    name: str | None = Field(default=None, max_length=128)
    description: str | None = Field(default=None)

    @field_validator("name")
    @classmethod
    def _normalize_name(cls, value: str | None) -> str | None:
        if value is None:
            return None
        stripped = value.strip()
        return stripped or None

    @field_validator("description")
    @classmethod
    def _normalize_description(cls, value: str | None) -> str | None:
        if value is None:
            return None
        stripped = value.strip()
        return stripped or None


class FaultTypeSuggestionAcceptResponse(BaseModel):
    suggestion: FaultTypeSuggestionResponse
    fault_type: FaultTypeResponse
